catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/DataListFragment.kt

```diff
@@ -389,11 +509,57 @@ class DataListFragment : Fragment(),
                     .apply()
                 indexAndSort()
             }
+            R.id.undo -> undoLastPerformedAction()
             else -> return super.onOptionsItemSelected(item)
         }
         return true
     }
 
+    private fun undoLastPerformedAction() {
+        if (lastPerformedAction != null) {
+            var alertTitle = ""
+            var alertMsg = ""
+
+            when (lastPerformedAction!!.actionType) {
```

Comment:
using !! increases risk of crashing, other ways like using elvis ?: are safer, please change